cli/tests/test_release_channel.py

```
@@ -0,0 +1,1955 @@
+        env=env,
+        capture_output=True,
+        text=True,
+        check=False,
+        timeout=SCRIPT_TIMEOUT_SECONDS,
+    )
+
+    assert completed.returncode != 0
+    assert not (tmp_path / "curl.log").exists()
+
+
+def test_release_workflow_advances_channel_only_after_immutable_custody() -> None:
```

Ruff (0.15.21)

[warning] Too many statements (54 > 50)

(PLR0915)